src/NFTBill.sol

```diff
@@ -39,6 +40,29 @@ contract NFTBill is ERC1155 {
         _mint(msg.sender, id, 1, '');
     }
 
+    function depositWithPermit(
+        address erc20,
+        uint96 value,
+        uint256 deadline,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) external {
+        if (value <= 0) revert ValueTooSmall();
+
```

Comment:
this is redundant now

Reply:
Thought short circuiting might help save some gas... removed the check :+1: @w1nt3r-eth